apps/evm/lib/evm/vm.ex

```diff
@@ -144,4 +144,41 @@ defmodule EVM.VM do
 
     {final_machine_state, n_sub_state, n_exec_env}
   end
+
+  def log(operation, machine_state) do
+    operation_string = Atom.to_string(operation.sym)
+      |> String.upcase()
+      |> String.pad_leading(8)
+
+    opcode_string = operation.id
+      |> :binary.encode_unsigned()
+      |> Base.encode16(case: :lower)
+      |> String.trim_leading("0")
+      |> String.pad_trailing(2)
+
+    program_counter = machine_state.program_counter + 1
+    program_counter_string = "0x#{program_counter}"
+      |> :binary.encode_unsigned()
+      |> Base.encode16(case: :lower)
+      |> String.trim_leading("0")
+      |> String.pad_trailing(3)
+
+    IO.puts "[#{program_counter_string}][#{operation_string}(0x#{opcode_string}) Gas Left: #{machine_state.gas})"
```

Comment:
Maybe we could use logger instead of puts

Reply:
This is formatted exactly the same as Parity's `evm-debug` output. That makes it really easy to compare output when debugging differences between parity and our implementation. I think we can change the formatting of `Logger` but that might change it globally. We should start to think about logging at some point though. It's be nice to have log levels like the other clients do at some point...